src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java

```diff
@@ -1430,22 +1423,16 @@ && ifResponseTypeIsPartOrFullAdmission(caseData)) {
                 .applicant1ResponseDeadline(getApplicant1ResponseDeadline(responseDate))
                 .businessProcess(BusinessProcess.ready(DEFENDANT_RESPONSE_SPEC));
 
-            if (caseData.getRespondent2() != null && caseData.getRespondent2Copy() != null) {
+            if (caseData.getRespondent2() != null) {
                 Party updatedRespondent2;
 
                 if (NO.equals(caseData.getSpecAoSRespondent2HomeAddressRequired())) {
                     updatedRespondent2 = caseData.getRespondent2().toBuilder()
                         .primaryAddress(caseData.getSpecAoSRespondent2HomeAddressDetails()).build();
                 } else {
-                    updatedRespondent2 = caseData.getRespondent2().toBuilder()
-                        .primaryAddress(caseData.getRespondent2Copy().getPrimaryAddress()).build();
+                    CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder();
+                    PersistDataUtils.persistFlagsForParties(caseData, oldCaseData, caseDataBuilder);
                 }
-
-                updatedData
-                    .respondent2(updatedRespondent2.toBuilder()
-                                     .flags(caseData.getRespondent2Copy().getFlags()).build())
-                    .respondent2Copy(null);
-                updatedData.respondent2DetailsForClaimDetailsTab(updatedRespondent2.toBuilder().flags(null).build());
```

Comment:
I feel like this persistFlagsForParties method should be here no due to the case flag issue?